service-workers/redirects/fragment.https.window.js

```diff
@@ -0,0 +1,35 @@
+// META: script=/html/canvas/resources/canvas-tests.js
+// META: script=../service-worker/resources/test-helpers.sub.js
+
+promise_test(async t => {
+  const reg = await service_worker_unregister_and_register(t, "resources/fragment-serviceworker.js", "resources/");
+  // Not add_cleanup as we want this to run after all tests
+  add_completion_callback(async () => await reg.unregister());
+  await wait_for_state(t, reg.installing, 'activated');
+  const frame = await with_iframe("resources/dummy.html");
+  const canvas = frame.contentDocument.body.appendChild(document.createElement("canvas"));
+  canvas.width = 100;
+  canvas.height = 50;
+  const ctx = canvas.getContext("2d");
+
+  // Changing this list requires corresponding changes in resources/fragment-serviceworker.js
+  [
+    { input: "?1#green", subtitle: "control" },
+    { input: "?2#green", subtitle: "request overrides" },
```

Comment:
It still feels wrong that this is not consistent with redirects.  It really feels like it should work the same way as:

https://github.com/web-platform-tests/wpt/blob/8602e9c9a168c25838bc4b808e3d213ce94ccac1/fetch/redirects/subresource-fragments.html#L19

I know we did this partly to avoid the issue with cache_storage, but I think that may be a mistake.  I'm starting to believe we should match redirect behavior here and sites would just need to be careful not to store responses with fragments in cache_storage if they care about multi-fragment to the same resource.